### PR TITLE
OUT-329 Activity Feed

### PR DESCRIPTION
**Pull request by arpandhakal**
### Tasks

- [Activity Feed](https://linear.app/copilotplatforms/issue/OUT-329/activity-feed)
- [API - Create / Update / Delete comments or subcomments](https://linear.app/copilotplatforms/issue/OUT-372/api-create-update-delete-comments-or-subcomments)

### Changes

- [x] created postcomment functionality from ui side
- [x] refactored comments' response type
- [x] created deletecomment functionality from ui side
- [x] integrated create, list, delete comments and replies functionality

### Testing Criteria

- [LOOM](https://www.loom.com/share/6271378ba5964154886efbc887a46d08)

### CODE REVIEW

File: src/components/inputs/CommentInput.tsx [hunk 1]
```diff
@@ -6,14 +6,32 @@ import { AttachmentIcon } from '@/icons'
...
+    }
+  }
   return (
     <Stack direction="row" columnGap={3} alignItems="flex-start">
       <Avatar alt="user" src={''} sx={{ width: '25px', height: '25px' }} />
       <CommentCardContainer>
-        <TapWriteCommentInput content={''} getContent={(content) => setDetail(content)} placeholder="Leave a comment..." />
+        <TapWriteCommentInput
+          content={detail}
+          getContent={(content) => setDetail(content)}
```
Comment (rrojan) [inline src/components/inputs/CommentInput.tsx:32]:
```suggestion
          getContent={setDetail}
```

File: src/components/cards/CommentCard.tsx [hunk 2]
```diff
@@ -62,27 +95,25 @@ export const CommentCard = ({ comment }: { comment: any }) => {
...
             <Stack direction="column" rowGap={3} key={item.id}>
               <CustomDivider />
               <Stack direction="row" columnGap={2} alignItems={'center'}>
-                <Avatar
-                  alt="user"
-                  src={comment?.iconImageUrl || comment?.avatarImageUrl}
-                  sx={{ width: '20px', height: '20px' }}
-                />
-                <BoldTypography>{item.details?.initiator}</BoldTypography>
+                <Avatar alt="user" src={''} sx={{ width: '20px', height: '20px' }} />
```
Comment (rrojan) [inline src/components/cards/CommentCard.tsx:105]: Is there a reason why src is empty string? Does it default to something else when empty?

File: src/components/cards/CommentCard.tsx [hunk 3]
```diff
@@ -53,7 +79,14 @@ export const CommentCard = ({ comment }: { comment: any }) => {
...
                   menuContent={
-                    <ListBtn content="Delete" handleClick={() => {}} icon={<TrashIcon />} contentColor="#CC0000" />
+                    <ListBtn
+                      content="Delete"
+                      handleClick={() => {
+                        setShowConfirmDeleteModal(true)
+                      }}
+                      icon={<TrashIcon />}
+                      contentColor="#CC0000"
+                    />
```
Comment (rrojan) [inline src/components/cards/CommentCard.tsx:89]: If possible use the global `theme` object instead of hardcoding color values

**Review by rrojan — CHANGES_REQUESTED** [review]

File: src/components/cards/CommentCard.tsx (hunk 2 above)
Comment (arpandhakal) [reply src/components/cards/CommentCard.tsx:105]: Yes. As per @aatbip, right now we dont get AvatarUrl from getInterUsers function from the copilot SDK. I thought once the copilot team gets to do this, I will add the urls.

**Review by rrojan — APPROVED** [review]
LGTM :trophy:

**Review by saugatmaharjan — APPROVED** [review]

**Pull request merged**